src/memory.rs

```diff
@@ -0,0 +1,53 @@
+use crate::constants::MEMORY_MAX;
+
+pub struct Memory {
+    memory: [u16; MEMORY_MAX as usize],
```

Comment:
This should probably be:
```suggestion
    memory: [u16; MEMORY_MAX as usize + 1],
```
It needs to fit all the `u16` addresses, which is `2^16`.

Reply:
Good catch 💀